cmd/kaidump/kaidump.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/openware/kaigara/pkg/config"
+	"github.com/openware/kaigara/pkg/vault"
+
+	"github.com/openware/pkg/ika"
+	"strings" // Needed to use Split
+	"gopkg.in/yaml.v3"
+)
+
+var cnf = &config.KaigaraConfig{}
+
+func initConfig() {
+	err := ika.ReadConfig("", cnf)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func getVaultService(appName string) *vault.Service {
+	return vault.NewService(cnf.VaultAddr, cnf.VaultToken, cnf.DeploymentID)
+}
+
+func main() {
+	// Parse flags
+	filepath := flag.String("a", "outputs.yaml", "Outputs file path to save secrets from vault")
+	flag.Parse()
+
+	// Initialize and write to Vault stores for every component
+	initConfig()
+	secretStore := getVaultService("global")
+	apps, err := secretStore.ListAppNames()
+	if err != nil {
+		panic(err)
+	}
+	scopes := os.Getenv("KAIGARA_SCOPES")
+	if scopes == "" {
+		panic("Undefined KAIGARA_SCOPES env")
+	}
+	scopesList := strings.Split(scopes, ",")
+	if len(scopesList) <= 0 {
+		panic("Wrong KAIGARA_SCOPES")
+	}
+	appMap := make(map[string]interface{})
+	for _, app := range apps {
+		scopeMap := make(map[string]interface{})
+		for _, scope := range scopesList {
+			err := secretStore.LoadSecrets(app, scope)
+			if err != nil {
+				panic(err)
+			}		
+			secrets, err := secretStore.GetSecrets(app, scope)
+			if err != nil {
+				panic(err)
+			}
+			scopeMap[scope] = secrets
+		}
+		appMap[app] = scopeMap
+	}
```

Comment:
Let's add newlines between significant code blocks

Reply:
Added new lines and comments details